### PR TITLE
Modularize datageneration

### PR DESCRIPTION
**Pull request by cmonez**
-Got rid of old data generation files
-Created a generalized "generateCSVdata" function where you specify your data shape and and the amount of records to create, as well as the starting documentId (in our case, were starting at 100)
-Created a single file where I run all of my data generation functionality
-Re-added headers to CSV files and specified in COPY function to omit when seeding the database to avoid errors with datatypes 
-Ran tests for data generation logic. 

@jeremylfchen @Nick-Zabalza

### CODE REVIEW

File: test/dataGeneration.test.js [hunk 1]
```diff
@@ -1,34 +1,70 @@
...
-  //
-  it('should count 50 lines in the attributes csv file', () => {});
-  //
-  it('should count 50 lines in the directions csv file', () => {});
-  //
-  it('should count 50 lines in the details csv file', () => {});
+
+  it('Should generate 4 objects of type string for the generateAttributes function', () => {
+    let attributesObject = generateAttributesShape();
+   attributesObject = attributesObject.split('", "')
```
Comment (jeremylfchen) [inline test/dataGeneration.test.js]: Needs 1 more space at the start of the line. Also lines 49-54 need semicolons

File: test/dataGeneration.test.js [hunk 2]
```diff
@@ -1,34 +1,70 @@
...
+  })
+
+  it('Should generate text for details', () => {
+    let details = generateDetailsShape();
+    details = [details]
+    expect(typeof details[0]).toBe('string');
+  })
+
+  it('Should generate text for directions', () => {
+    let directions = generateDetailsShape();
```
Comment (jeremylfchen) [inline test/dataGeneration.test.js]: I think you meant to call `generateDirectionsShape` here?

**Review by jeremylfchen — COMMENTED** [review]
Hi Eric,

Looks good over all. I like how you made you data generation functions more modular and cut down the number of files required from 4 to now just 1.

A few notes. Your test file for data gen doesnt seem to call `generateDirectionsShape`. I think it was a copy and past mistake. I also made some comments on the styling lol.

My main question is for you tables. It seems like `attributes`, `details`, `descriptions`, and `directions` are all a 1 to 1 to 1 to 1 relationship correct? But you seem to have `details`, `attributes`, and `directions` all reference `descriptions` instead of the other way around. How do you plan to make your queries? To my understanding, it seems you would have to make 4 separate queries to get all the required data instead of having `descriptions` store the index/id of the other 3 tables and just execute some chained inner joins using 1 query.

Let me know what you think

also all you tests pass on my end, will need to test again once you invoke `generateDirectionsShape` in the last test

File: test/dataGeneration.test.js (hunk 2 above)
Comment (cmonez) [reply test/dataGeneration.test.js]: I actually meant to call Shape, forgot to chagne the test suite name.

**Comment by cmonez** [comment]
> Hi Eric,
> 
> Looks good over all. I like how you made you data generation functions more modular and cut down the number of files required from 4 to now just 1.
> 
> A few notes. Your test file for data gen doesnt seem to call `generateDirectionsShape`. I think it was a copy and past mistake. I also made some comments on the styling lol.
> 
> My main question is for you tables. It seems like `attributes`, `details`, `descriptions`, and `directions` are all a 1 to 1 to 1 to 1 relationship correct? But you seem to have `details`, `attributes`, and `directions` all reference `descriptions` instead of the other way around. How do you plan to make your queries? To my understanding, it seems you would have to make 4 separate queries to get all the required data instead of having `descriptions` store the index/id of the other 3 tables and just execute some chained inner joins using 1 query.
> 
> Let me know what you think
> 
> also all you tests pass on my end, will need to test again once you invoke `generateDirectionsShape` in the last test

Yeah i'll be using innerjoins. Storing the 3 ids in one table sounds like it'd take up a lot of space, but i'll think about it

**Pull request merged**